velox/dwio/parquet/tests/ParquetTableScanTest.cpp

```diff
@@ -25,15 +25,91 @@ using namespace facebook::velox;
 using namespace facebook::velox::exec;
 using namespace facebook::velox::exec::test;
 
-static const std::string kWriter = "ParquetTableScanTest.Writer";
-
 class ParquetTableScanTest : public HiveConnectorTestBase {
  protected:
   using OperatorTestBase::assertQuery;
 
   void SetUp() override {
     HiveConnectorTestBase::SetUp();
     parquet::registerParquetReaderFactory();
+
+    loadData();
+  }
+
+  void assertSelect(
+      const std::vector<std::string>&& outputColNames,
+      const std::string& sql) {
+    std::vector<std::shared_ptr<const Type>> types;
+    for (auto colName : outputColNames) {
+      types.push_back(rowTypes->findChild(colName));
+    }
+
+    auto plan =
+        PlanBuilder()
+            .tableScan(
+                ROW(const_cast<::std::vector<std::string>&&>(outputColNames),
+                    std::move(types)))
+            .planNode();
+
+    assertQuery(plan, splits, sql);
+  }
+
+  void assertSelectWithFilter(
+      const std::vector<std::string>&& outputColNames,
+      common::test::SubfieldFilters filters,
+      const std::string& sql) {
+    std::vector<std::shared_ptr<const Type>> types;
+    for (auto colName : outputColNames) {
+      types.push_back(rowTypes->findChild(colName));
+    }
+    auto rowType =
+        ROW(const_cast<::std::vector<std::string>&&>(outputColNames),
+            std::move(types));
+
+    auto plan = PlanBuilder()
+                    .tableScan(
+                        rowType,
+                        makeTableHandle(std::move(filters)),
+                        allRegularColumns(rowType))
+                    .planNode();
+
+    assertQuery(plan, splits, sql);
+  }
+
+  void assertSelectWithAgg(
+      const std::vector<std::string>&& outputColNames,
+      const std::vector<std::string>& aggregates,
+      const std::vector<ChannelIndex>& groupingKeys,
+      const std::string& sql) {
+    std::vector<std::shared_ptr<const Type>> types;
+    for (auto colName : outputColNames) {
+      types.push_back(rowTypes->findChild(colName));
+    }
+
+    auto plan =
+        PlanBuilder()
+            .tableScan(
+                ROW(const_cast<::std::vector<std::string>&&>(outputColNames),
+                    std::move(types)))
+            .singleAggregation(groupingKeys, aggregates)
+            .planNode();
+
+    assertQuery(plan, splits, sql);
+  }
+
+ private:
+  void loadData() {
```

Comment:
Each test will likely have a different schema. There is no value in making this a class member if we cannot reuse it across tests. Should we move this back to the test body?

Reply:
@majetideepak This makes sense. But I still kept `rowTypes_` and `splits_` as the class member, since all the `testSelectxxx()` functions need to use them. I made `loadTable()` to take input parameters to identify which table the test wants to load:
```
  void loadData(
      const std::string& filePath,
      std::shared_ptr<const RowType> rowTypes,
      RowVectorPtr data) {
    splits_ = {makeSplit(filePath)};
    rowTypes_ = rowTypes;
    createDuckDbTable({data});
  }
```
Then in the test it's like
```
TEST_F(ParquetTableScanTest, basic) {
  loadData(
      getExampleFilePath("sample.parquet"),
      ROW({"a", "b"}, {BIGINT(), DOUBLE()}),
      makeRowVector(
          {"a", "b"},
          {
              makeFlatVector<int64_t>(20, [](auto row) { return row + 1; }),
              makeFlatVector<double>(20, [](auto row) { return row + 1; }),
          }));
```
How do you think?

Reply:
This looks good. Thanks!